cargo: Fix compiling on rust 1.66

cathay4t opened the pull request:
We need to support compiling nmstate on rust 1.66 (the version shipped
by RHEL 9.2).

Except the `--ignore-rust-version` compiling option, we need to pin the
version of `ctrlc` crate to 3.4.2 to pass the compiling as its later version use nix
0.28 which requires rust 1.69+.

Included CI test case to ensure this.

liangwen12year commented on .github/workflows/main.yml:
```diff
@@ -169,6 +169,17 @@ jobs:
+      run: rustup install 1.66
+
+    - name: Build on rust 1.66
+      run: cd rust && cargo +1.66 build --ignore-rust-version
```
Can we drop the option `--ignore-rust-version`? IMHO, if the rust compiler version is older than the required rust version, we should not proceed to build the target. 

```
       --ignore-rust-version
           Build the target even if the selected Rust compiler is older than
           the required Rust version as configured in the project’s
           rust-version field.
```

cathay4t replied:
The `ignore-rust-version` is for other crates.

liangwen12year commented:
About the CI failure on `fed-nm_stable-integ_tier1`, 

```
error: failed to select a version for the requirement `nispor = "^1.2.18"`
candidate versions found which didn't match: 1.2.17
```

I notice that the nispor version is already updated to 1.2.18 in Fedora, https://src.fedoraproject.org/rpms/nispor/c/9bd900db0fdb45aab259641da31d950a7b3b0c74?branch=rawhide.
Maybe the quay.io image needs to be rebuilt?

cathay4t commented:
> About the CI failure on `fed-nm_stable-integ_tier1`,
> 
> ```
> error: failed to select a version for the requirement `nispor = "^1.2.18"`
> candidate versions found which didn't match: 1.2.17
> ```
> 
> I notice that the nispor version is already updated to 1.2.18 in Fedora, https://src.fedoraproject.org/rpms/nispor/c/9bd900db0fdb45aab259641da31d950a7b3b0c74?branch=rawhide. Maybe the quay.io image needs to be rebuilt?

Fixed.

cathay4t committed:
cargo: Fix compiling on rust 1.66

We need to support compiling nmstate on rust 1.66 (the version shipped
by RHEL 9.2).

Except the `--ignore-rust-version` compiling option, we need to pin the
version of `ctrlc` crate to 3.4.2 to pass the compiling as it later
version use nix 0.28 which requires rust 1.69+.

Included CI test case to ensure this.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Force merging as github CI is down.

 https://github.com/nmstate/nmstate/actions/runs/8815898674 confirmed the 1.66 compiling is working.
